fairscale/optim/oss.py

```diff
@@ -140,13 +140,6 @@ def partition_parameters(self) -> List[List[dict]]:
                     param_group_rank["params"] = params
                     self._partition_parameters[rank].append(param_group_rank)
 
```

Comment:
changing the logic: if a rank has an empty shard, it will be taken care of bucket-wise, no more asserting since that prove fragile on fb infra (the job can be compiled without the asserts)

Reply:
Just to clarify: this happens when one of the nodes does not have any params assigned to it? Thats an empty shard right?